Added a new App that is Change-O for singularity.

bcorrie opened the pull request:
I have created a Tapis App that works with Change-O (Immcantation) using singularity.

Discuss development on this app/branch through this pull request.

bcorrie committed:
Added a new App that is Change-O for singularity.

bcorrie commented:
I got an App built, I can load the App, I created a Job, using the App, and it ran without errors.

The code currently has an error. In changeo.sh it has an error, it tries to run "run_changeo_workflow" (line 56) when I changed changeo_common.sh to have a function called run_workflow(). This needs to be fixed... Over to the European hackathon'ers to try and improve on where I got to if you want...

bcorrie committed:
Cleaned up some stuff that isn't needed for this app from the toy app.

bcorrie committed:
Changed it so that the App uses a file from a data directory. This is
how things would normally work in that you give the App a URL for a
Tapis file somewhere (possibly anywhere, even on an iReceptor repository).

The App then stages this file to the execution system for processing.

bcorrie committed:
Got rid of the test data file. Don't use it and it is large so
don't want to transfer it.

bcorrie committed:
Added a shell script to update important files into the App staging
area for this App.

bcorrie committed:
Cleaned up and got rid of a bunch of stuff from the old Toy App.

Very simple now, just runs singularity to generate the clones for
a single AIRR TSV file. It seems to work!

bcorrie committed:
Removed test file, not needed.

bcorrie commented:
Working Change-O app I believe! 

Inputs:
- Tapis URL for the Singularity image
- Tapis URL for an AIRR TSV file (file.tsv)

Output:
- Error and out files
- AIRR TSV file with added clone_id column with name file_clone-pass.tsv

bcorrie commented:
Todo:
- Parameter passing (to control how Change-O processes the file)

bcorrie committed:
Add a submit example and some docs for usefule commands.

bcorrie committed:
Added a hello-world app, started on a cedar implementation for toy app.

bcorrie committed:
Merge branch 'change-o' of https://github.com/ireceptor-plus/tapis into change-o

bcorrie committed:
Update to hello-world.json - app create working.

bcorrie committed:
Cleanup...

bcorrie committed:
First working version of hello world app and job!!!

bcorrie committed:
Minor changes cleaning up...

bcorrie committed:
Cleaned up some more bits - made all files more "minimal"

bcorrie committed:
Added storage and compute system descriptions for the Compute Canada
cedar ARC platform.

bcorrie commented:
Got a basic hello world app running using Tapis on cedar.computecanada.ca, using the group account for the iReceptor gateway...

bcorrie committed:
Initial test verion.

bcorrie committed:
Initial version - not so common between cedar and TACC.

bcorrie committed:
Get rid of some TACC specific module loads.

bcorrie committed:
Removed extra stuff not required by cedar.

bcorrie committed:
First attempt at working version on cedar.

bcorrie committed:
Fixes and added common file... Not so common at the moment.

bcorrie committed:
Updates to sort out local environment problems.

Commit:
Added path to singularity mount point

We need to be able to find the file inside the container.

bcorrie committed:
Added number of processor handling...

bcorrie committed:
Removed use of airr_file parameter

bcorrie committed:
Minor changes to processors and memory for testing.

bcorrie committed:
Change memory specification

bcorrie committed:
Added a temporary default memory allocation per core.

bcorrie committed:
Changed to new node specification

bcorrie committed:
Added a VDJBase app directory.

bcorrie committed:
Fixes for VDJBase pipeline versions.

bcorrie committed:
Fix path

bcorrie committed:
Singularity directory fixes.

bcorrie committed:
Remove versions for now...

bcorrie committed:
Stupid bug!

bcorrie committed:
Added a sample parameter to the app.

bcorrie committed:
Got rid of old unused "demo" parameters from the app

bcorrie commented:
Working Change-O and VDJBase Tapis based Apps working on Compute Canada's cedar ARC platform.

schristley commented on apps/vdjbase-singularity/0.1/cedar/vdjbase.json:
```diff
@@ -0,0 +1,116 @@
+      }
+    },
+    {
+      "id": "sample_name",
```
is this a custom field? does it map to an existing AIRR field? I would guess `repertoire_id`?

bcorrie replied:
Not 100% sure. Pretty sure it is not sample_id, but we don't have enough experience with VDJBase to know for sure how broad an input AIRR TSV file can be. When inferring alleles I am not sure repertoire_id would be sufficient, because I assume it doesn't do TCR and IGH at the same time??? 

Currently it is a required parameter by VDJBase (or at least a helpful one) that is supplied by the user to group the processed output files.

bcorrie replied:
I am guessing this would need the [repertoire_id, sample_processing_id, data_processing_id]

schristley commented on apps/vdjbase-singularity/0.1/cedar/upload-bundle.sh:
```diff
@@ -0,0 +1,23 @@
+#
+TOOL=changeo-singularity
+SYSTEM=ls5
```
cedar?

bcorrie replied:
I don't use this file in the current App, I am using Git to stage changes to the deployment directory on cedar. It should be changed, yes...

bcorrie replied:
Fixed on my latest commit, although still not using these files.

schristley commented on apps/vdjbase-singularity/0.1/cedar/update-bundle.sh:
```diff
@@ -0,0 +1,15 @@
+#
+TOOL=changeo-singularity
+SYSTEM=ls5
```
cedar?

bcorrie replied:
Same as above, not used, but didn't want to delete.

bcorrie committed:
Changed system name in bundle management scripts.

bcorrie commented:
@schristley want to merge this into master.

Changes in summary are:

- Added cedar directories to most apps
- Added a changeo-singularity app that basically works
- Added a vdjbase-singularity app that basically works

They need more work, but would like to merge these in to master and then make improvements individually. Some of these changes go way back to our first hackathon...

bcorrie commented:
Let me know if you have any objections...

schristley commented:
@bcorrie good to merge, ls5 is dead but we can keep the files for history, I will need to port to stampede2 if want to use on TACC.
